tests/manage/pv_services/pvc_snapshot/test_small_files_pvc_cloned_status.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import random
+import os
+import urllib.request
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.framework.testlib import bugzilla
+from ocs_ci.ocs.node import get_worker_nodes
+from ocs_ci.ocs.resources import pod, pvc
+from ocs_ci.helpers import helpers
+from ocs_ci.framework.testlib import (
+    E2ETest,
+    tier1,
+    polarion_id,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@polarion_id("OCS-3946")
+class TestPvcClonedStatusOfSmallSizeFiles(E2ETest):
+    """
+    Tests to verify restored PVC status bound or not if
+    files size are smaller.
+    """
+
+    @bugzilla("2039265")
+    def test_pvc_status_after_clone(
+        self, pvc_factory, teardown_factory, snapshot_factory
+    ):
+        """
+        1. Create 1GiB PVC
+        2. Attach PVC to an application pod
+        3. Download Linux kernel and untar it to the directory where PVC is mounted
+        4. Create a new PVC from the snapshot
+        5. Take a snapshot of the PVC.
+        6. Create a new PVC out of that snapshot.
+        7. Verify PVC state
+
+        Args:
+            pvc_factory: A fixture to create new pvc
+            teardown_factory: A fixture to destroy objects
+            snapshot_factory: A fixture to create new snapshot
+        """
+        pvc_size = 1
+        worker_nodes_list = get_worker_nodes()
+        kernel_url = "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.4.tar.xz"
+        download_path = "tmp"
+        dir_path = os.path.join(os.getcwd(), download_path)
+        file_path = os.path.join(dir_path, "linux-4.4.tar.xz")
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        urllib.request.urlretrieve(kernel_url, file_path)
+
+        # Create a RWX Cephfs PVC
+        pvc_obj = pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            access_mode=constants.ACCESS_MODE_RWX,
+            status=constants.STATUS_BOUND,
+            size=pvc_size,
+            size_unit="Gi",
+        )
+
+        # Confirm PVCs are Bound
+        log.info("Verifying the CephFS PVC are Bound")
+        helpers.wait_for_resource_state(
+            resource=pvc_obj, state=constants.STATUS_BOUND, timeout=240
+        )
+        pvc_obj.reload()
+        log.info("Verified: CephFS PVC are Bound")
+
+        print("PVC values:", vars(pvc_obj))
+
+        # Create a pod on a particular node
+        selected_node = random.choice(worker_nodes_list)
+        log.info(f"Creating a pod on node: {selected_node} with pvc {pvc_obj.name}")
+
+        pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHFILESYSTEM,
+            pvc_name=pvc_obj.name,
+            namespace=pvc_obj.namespace,
+            node_name=selected_node,
+            pod_dict_path=constants.PERF_POD_YAML,
+        )
+        teardown_factory(pod_obj)
+        log.info("Pod Teardown complete")
```

Comment:
This log.info is not needed since pod teardown won't happen until your test completes executing.

Reply:
Okay